address-space-controller/src/main/java/io/enmasse/controller/CreateController.java

```diff
@@ -163,47 +168,46 @@ public AddressSpace reconcileActive(AddressSpace addressSpace) throws Exception
                     return addressSpace;
                 }
                 KubernetesList resourceList = new KubernetesListBuilder()
-                        .addAllToItems(infraResourceFactory.createInfraResources(addressSpace, desiredInfraConfig))
+                        .addAllToItems(infraResourceFactory.createInfraResources(addressSpace, desiredInfraConfig, desiredConfig.getAuthenticationServiceSettings()))
                         .build();
                 addAppliedInfraConfigAnnotation(resourceList, desiredInfraConfig);
-                addAppliedPlanAnnotation(resourceList, addressSpace.getSpec().getPlan());
+                addAppliedConfigAnnotation(resourceList, desiredConfig);
 
                 log.info("Upgrading address space {}", addressSpace);
 
                 kubernetes.apply(resourceList,desiredInfraConfig.getUpdatePersistentVolumeClaim());
                 eventLogger.log(AddressSpaceUpgraded, "Upgraded address space", Normal, ControllerKind.AddressSpace, addressSpace.getMetadata().getName());
                 InfraConfigs.setCurrentInfraConfig(addressSpace, desiredInfraConfig);
-                addressSpace.putAnnotation(AnnotationKeys.APPLIED_PLAN, addressSpace.getSpec().getPlan());
+                AppliedConfig.setCurrentAppliedConfig(addressSpace, desiredConfig);
             } else {
                 log.info("Version of desired config ({}) does not match controller version ({}), skipping upgrade", desiredInfraConfig.getVersion(), version);
             }
-        } else if (!addressSpace.getSpec().getPlan().equals(appliedPlan)) {
+        } else if (!desiredConfig.equals(appliedConfig)) {
```

Comment:
The code paths for unequal infra configs and unequal applied configs now look similar.  The only material difference is the infraconfig version check - which I'm not sure remains important.  It makes me wonder if the `AppliedConfig` object should carry the infraconfig object too, allowing the infra config annotation and the `else if` to be removed.

Reply:
I agree they should be merged. However, I would perhaps like to do this separately to isolate the effects.